rules/aws_resource_missing_tags.go

```diff
@@ -77,29 +148,79 @@ func (r *AwsResourceMissingTagsRule) Check(runner tflint.Runner) error {
 		}
 
 		resources, err := runner.GetResourceContent(resourceType, &hclext.BodySchema{
-			Attributes: []hclext.AttributeSchema{{Name: tagsAttributeName}},
+			Attributes: []hclext.AttributeSchema{
+				{Name: tagsAttributeName},
+				{Name: providerAttributeName},
+			},
 		}, nil)
 		if err != nil {
 			return err
 		}
 
+		if resources.IsEmpty() {
+			continue
+		}
+
 		for _, resource := range resources.Blocks {
-			if attribute, ok := resource.Body.Attributes[tagsAttributeName]; ok {
-				logger.Debug("Walk `%s` attribute", resource.Labels[0]+"."+resource.Labels[1]+"."+tagsAttributeName)
-				wantType := cty.Map(cty.String)
-				err := runner.EvaluateExpr(attribute.Expr, func(resourceTags map[string]string) error {
-					r.emitIssue(runner, resourceTags, config, attribute.Expr.Range())
+			providerAlias := "default"
+			// Override the provider alias if defined
+			if val, ok := resource.Body.Attributes[providerAttributeName]; ok {
+				provider, diagnostics := aws.DecodeProviderConfigRef(val.Expr, providerAlias)
+				providerAlias = provider.Alias
+
+				if _, hasProvider := providerTagsMap[providerAlias]; !hasProvider {
+					errString := fmt.Sprintf(
+						"The aws provider with alias \"%s\" doesn't exist.",
+						providerAlias,
+					)
+					logger.Error("Error querying provider tags: %s", errString)
+					runner.EmitIssue(r, errString, *provider.AliasRange)
```

Comment:
Emitting this error as an issue in the `aws_resource_missing_tags` rule is inappropriate.
Perhaps we should simply return an error in this case.

Reply:
Fixed!

Changed a little bit how the tests were done, now also checks for a RaiseErr attribute if defined